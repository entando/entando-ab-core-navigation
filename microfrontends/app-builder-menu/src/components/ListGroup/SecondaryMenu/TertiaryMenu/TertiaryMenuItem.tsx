import { useContext } from 'react';
import styled from 'styled-components';
import MenuUIContext from '../../../MenuUIContext';

const StyledLink = styled.a<{ isActive: boolean }>`
  width: 100%;
  background-color: transparent;
  background-color: ${({ isActive }) => (isActive ? '#4D5257' : 'transparent')};
  color: ${({ isActive }) => (isActive ? '#fff' : '#d1d1d1')};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  line-height: 26px;
  padding: 5px 10px 5px 8px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  margin-top: 4px;

  &:hover {
    font-weight: 600;
    color: #fff;
    background-color: #4d5257;
  }
`;

const StyledLabel = styled.span`
  display: block;
  flex: 1;
  max-width: none;
  overflow: hidden;
  text-overflow: ellipsis;
`;

interface Props {
  id: string;
  label: string;
  onClick?: () => void;
}

export default function TertiaryMenuItem(props: Props): JSX.Element {
  const { id, label, onClick } = props;

  const { activeTertiaryMenuItemId, setActiveTertiaryMenuItemId } =
    useContext(MenuUIContext);

  const isActive = activeTertiaryMenuItemId === id;

  const handleClick = (): void => {
    setActiveTertiaryMenuItemId(id);
    onClick && onClick();
  };

  return (
    <StyledLink isActive={isActive} onClick={handleClick}>
      <StyledLabel>{label}</StyledLabel>
    </StyledLink>
  );
}
