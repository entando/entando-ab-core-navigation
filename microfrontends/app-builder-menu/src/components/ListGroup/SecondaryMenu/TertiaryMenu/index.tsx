import styled from 'styled-components';

const StyledSecondaryTertiaryMenu = styled.div<{ isVisible: boolean }>`
  background: #292e34;
  border: 1px solid #292e34;
  border-bottom: none;
  border-top: none;
  bottom: 0;
  display: block;
  left: 400px;
  overflow-x: hidden;
  overflow-y: auto;
  position: fixed;
  top: 0px;
  opacity: 0;
  visibility: hidden;
  ${({ isVisible }) => (isVisible ? `opacity: 1; visibility: visible;` : ``)}
  width: 200px;
  z-index: 1030;
`;

const StyledSecondaryListGroup = styled.ul`
  margin-top: 25px;
  padding-left: 22px;
  padding-right: 0px;
`;

interface Props {
  isOpen: boolean;
  children?: React.ReactNode;
}

export default function TertiaryMenu(props: Props): JSX.Element {
  const { isOpen, children } = props;
  return (
    <StyledSecondaryTertiaryMenu isVisible={isOpen}>
      <StyledSecondaryListGroup>{children}</StyledSecondaryListGroup>
    </StyledSecondaryTertiaryMenu>
  );
}
