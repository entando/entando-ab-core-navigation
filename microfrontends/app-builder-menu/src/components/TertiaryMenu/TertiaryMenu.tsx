import styled from 'styled-components';
import { COLORS } from './../theme';

const StyledSecondaryTertiaryMenu = styled.div<{ isVisible: boolean }>`
  background: ${COLORS.primary};
  border: 1px solid ${COLORS.primary};
  border-bottom: none;
  border-top: none;
  bottom: 0;
  display: block;
  left: 430px;
  overflow-x: hidden;
  overflow-y: auto;
  position: fixed;
  top: 60px;
  opacity: 0;
  visibility: hidden;
  ${({ isVisible }) => (isVisible ? `opacity: 1; visibility: visible;` : ``)}
  width: 215px;
  z-index: 1030;
`;

const StyledSecondaryListGroup = styled.ul`
  margin-top: 25px;
  padding-left: 22px;
  padding-right: 0px;
`;

interface Props {
  isOpen: boolean;
  children?: React.ReactNode;
}

export function TertiaryMenu(props: Props): JSX.Element {
  const { isOpen, children } = props;
  return (
    <StyledSecondaryTertiaryMenu isVisible={isOpen} data-id="menu">
      <StyledSecondaryListGroup>{children}</StyledSecondaryListGroup>
    </StyledSecondaryTertiaryMenu>
  );
}
