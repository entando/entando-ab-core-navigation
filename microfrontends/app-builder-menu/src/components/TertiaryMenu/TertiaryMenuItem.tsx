import { useContext } from 'react';
import styled from 'styled-components';
import { MenuUIContext } from './../MenuUIContext';
import { COLORS } from './../theme';

const StyledLink = styled.a<{ isActive: boolean }>`
  width: 100%;
  background-color: transparent;
  background-color: ${({ isActive }) =>
    isActive ? COLORS.secondary : 'transparent'};
  color: ${({ isActive }) => (isActive ? COLORS.white : COLORS.text)};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-family: 'Open Sans', sans-serif;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  line-height: 26px;
  padding: 5px 10px 5px 8px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  margin-top: 4px;

  &:hover {
    font-weight: 600;
    color: ${COLORS.white};
    background-color: ${COLORS.secondary};
  }
`;

const StyledLabel = styled.span`
  display: block;
  flex: 1;
  max-width: none;
  overflow: hidden;
  text-overflow: ellipsis;
`;

interface Props {
  id: string;
  dataId?: string;
  label: string;
  onClick?: () => void;
  href?: string;
  className?: string;
  target?: string;
  rel?: string;
}

export function TertiaryMenuItem(props: Props): JSX.Element {
  const {
    id,
    dataId,
    label,
    onClick,
    href,
    target,
    rel,
    className = ''
  } = props;

  const {
    activeTertiaryMenuItemId,
    setActiveTertiaryMenuItemId,
    setTertiaryMenuOpen
  } = useContext(MenuUIContext);

  const isActive = activeTertiaryMenuItemId === id;

  const handleClick = (): void => {
    setActiveTertiaryMenuItemId(id);
    setTertiaryMenuOpen(false);
    onClick && onClick();
  };

  return (
    <StyledLink
      isActive={isActive}
      onClick={handleClick}
      href={href}
      className={className}
      rel={rel}
      target={target}
      data-id={dataId}
    >
      <StyledLabel>{label}</StyledLabel>
    </StyledLink>
  );
}
