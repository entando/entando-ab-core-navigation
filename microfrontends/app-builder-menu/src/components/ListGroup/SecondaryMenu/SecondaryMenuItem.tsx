import styled from 'styled-components';
import RightArrowIcon from '../../Icons/RightArrowIcon';
import TertiaryMenu from './TertiaryMenu';

const StyledLink = styled.a<{ isActive: boolean }>`
  width: 100%;
  background-color: transparent;
  background-color: ${({ isActive }) => (isActive ? '#4D5257' : 'transparent')};
  color: ${({ isActive }) => (isActive ? '#fff' : '#d1d1d1')};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  line-height: 26px;
  padding: 5px 10px 5px 8px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  margin-top: 4px;

  &:hover {
    font-weight: 600;
    color: #fff;
    background-color: #4d5257;
  }
`;

const StyledLabel = styled.span`
  display: block;
  flex: 1;
  max-width: none;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const StyledSecondaryMenuItem = styled.li`
  display: flex;
`;

interface Props {
  isActive: boolean;
  hasChildren: boolean;
}

export default function SecondaryMenuItem(props: Props): JSX.Element {
  const { isActive, hasChildren } = props;
  return (
    <StyledSecondaryMenuItem>
      <StyledLink isActive={isActive}>
        <StyledLabel>Dashboard</StyledLabel>
        {hasChildren && <RightArrowIcon />}
      </StyledLink>
      <TertiaryMenu isOpen />
    </StyledSecondaryMenuItem>
  );
}
