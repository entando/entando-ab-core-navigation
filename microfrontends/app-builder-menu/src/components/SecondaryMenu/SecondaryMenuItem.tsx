import { SyntheticEvent, useContext } from 'react';
import styled from 'styled-components';
import { RightArrowIcon } from './../Icons/RightArrowIcon';
import { MenuUIContext } from './../MenuUIContext';
import { TertiaryMenu } from '../TertiaryMenu/TertiaryMenu';
import { COLORS } from './../theme';

const StyledLink = styled.a<{ isActive: boolean }>`
  width: 100%;
  background-color: transparent;
  background-color: ${({ isActive }) =>
    isActive ? COLORS.secondary : 'transparent'};
  color: ${({ isActive }) => (isActive ? COLORS.white : COLORS.text)};
  cursor: pointer;
  display: block;
  font-size: 14px;
  font-family: 'Open Sans', sans-serif;
  font-weight: ${({ isActive }) => (isActive ? '600' : '400')};
  line-height: 26px;
  padding: 5px 10px 5px 8px;
  position: relative;
  white-space: nowrap;
  text-decoration: none;
  display: flex;
  margin-top: 4px;

  &:hover {
    font-weight: 600;
    color: ${COLORS.white};
    background-color: ${COLORS.secondary};
  }
`;

const StyledLabel = styled.span`
  display: block;
  flex: 1;
  max-width: none;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const StyledSecondaryMenuItem = styled.li`
  display: flex;
`;

interface Props {
  id: string;
  dataId: string;
  children?: React.ReactNode;
  label: string;
  onClick?: () => void;
  href?: string;
  className?: string;
}

export function SecondaryMenuItem(props: Props): JSX.Element {
  const {
    id,
    dataId,
    children,
    label,
    onClick,
    href,
    className = '',
  } = props;

  const {
    activeSecondaryMenuItemId,
    setActiveSecondaryMenuItemId,
    tertiaryMenuOpen,
    setTertiaryMenuOpen
  } = useContext(MenuUIContext);

  const hasChildren = !!children;
  const isActive = activeSecondaryMenuItemId === id;

  const handleClick = (e: SyntheticEvent): void => {
    e.stopPropagation()

    setActiveSecondaryMenuItemId(id);
    if (hasChildren) {
      setTertiaryMenuOpen(true);
    } else {
      onClick && onClick();
    }
  };

  return (
    <>
      <StyledSecondaryMenuItem
        onClick={handleClick}
        className={className}
        data-id={dataId}
      >
        <StyledLink isActive={isActive} href={href}>
          <StyledLabel>{label}</StyledLabel>
          {hasChildren && <RightArrowIcon />}
        </StyledLink>
        {hasChildren && isActive && tertiaryMenuOpen && (
          <TertiaryMenu isOpen>{children}</TertiaryMenu>
        )}
      </StyledSecondaryMenuItem>
    </>
  );
}
