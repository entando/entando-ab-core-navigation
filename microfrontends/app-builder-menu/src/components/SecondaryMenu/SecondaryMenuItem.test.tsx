import React from 'react';
import renderer from 'react-test-renderer';
import { TertiaryMenuItem } from '../TertiaryMenu/TertiaryMenuItem';

import { SecondaryMenuItem } from './SecondaryMenuItem';

test('renders SecondaryMenuItem without children', () => {
  const tree = renderer
    .create(<SecondaryMenuItem id="item1" dataId="item1" label="Item 1" onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders SecondaryMenuItem with children', () => {
  const tree = renderer
    .create(
      <SecondaryMenuItem id="item1" dataId="item1" label="Item 1" onClick={() => {}}>
        <TertiaryMenuItem id="child1" label="Child 1" />
      </SecondaryMenuItem>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders SecondaryMenuItem with href', () => {
  const tree = renderer
    .create(
      <SecondaryMenuItem
        id="item1"
        dataId="item1"
        label="Item 1"
        onClick={() => {}}
        href="http://localhost:8080"
      >
        <TertiaryMenuItem id="child1" label="Child 1" />
      </SecondaryMenuItem>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
