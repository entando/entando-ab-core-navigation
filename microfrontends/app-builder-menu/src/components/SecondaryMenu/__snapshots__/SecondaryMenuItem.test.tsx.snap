// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SecondaryMenuItem with children 1`] = `
<li
  className="sc-gKXOVf YVohq"
  data-id="item1"
  onClick={[Function]}
>
  <a
    className="sc-eCYdqJ aNaTi"
  >
    <span
      className="sc-jSMfEi KDqqR"
    >
      Item 1
    </span>
    <svg
      fill="none"
      height={20}
      stroke="#979797"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="9 18 15 12 9 6"
      />
    </svg>
  </a>
</li>
`;

exports[`renders SecondaryMenuItem with href 1`] = `
<li
  className="sc-gKXOVf YVohq"
  data-id="item1"
  onClick={[Function]}
>
  <a
    className="sc-eCYdqJ aNaTi"
    href="http://localhost:8080"
  >
    <span
      className="sc-jSMfEi KDqqR"
    >
      Item 1
    </span>
    <svg
      fill="none"
      height={20}
      stroke="#979797"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="9 18 15 12 9 6"
      />
    </svg>
  </a>
</li>
`;

exports[`renders SecondaryMenuItem without children 1`] = `
<li
  className="sc-gKXOVf YVohq"
  data-id="item1"
  onClick={[Function]}
>
  <a
    className="sc-eCYdqJ aNaTi"
  >
    <span
      className="sc-jSMfEi KDqqR"
    >
      Item 1
    </span>
  </a>
</li>
`;
